### ISSUE TITLE
 Embedded YouTube, Webarchive, other Search engines 

### ISSUE THREAD

**Issue by gitut2007** [ddawson/Link-Tools#2]
Hi again, 
I apologize I didn't answer before in https://github.com/ddawson/Link-Tools/issues/1 , I thought it was all clear so I didn't check again, and I don't receive alerts from answers.

Yes, I meant embedded videos, so to 'copy their links into normal/short version YT links'
> you can usually use the linked title that appears at the top when you mouse over it. Link Tools is able to work with that.

I don't have an example right now, but when one doesn't have javascript enabled for YT [so actually one cannot view the video embedded, one only see the link] in a page, one only can copy one version of the link  sort of https...embeded....yt.com , no mouse over title option.

>  -copy/bookmark webarchive.org links into same but with date/hour taken
>  out, so one can visit the bookmark later at a more recent date.
> So, to the date selection pages, right?

I mean sort of convert https://web.archive.org/web/20190222375734/https://example.com/article/ into https://web.archive.org/web/https://example.com/article/ , which eg if bookmarked [or opened from copied link] would produce the last version of the page, maybe more recent than the date taken out.

> have to go to the settings, find the switch "Use POST vs GET", and change it to off, in order to be able to get search links from the site.

I guess maybe it would be less hassle [when needed I would see both ways to see...] to just copy and manually delete the extra stuff [specially if using search pages (Strtpag or other) without javascript and without cookies],
EDIT [now I see that in Strtpag I can copy clean links without the addon [I don't know if back then it was different or I hadn't realized it, I don't use Strtpag as my default ...}, anyway, in general, it stands as a suggestion to not limit the addon to G [for people that might prefer/need not to use them for searches, if in some case it's harder to copy links, so be it.]

Thank you.

**Comment by ddawson** [comment]
> I thought it was all clear so I didn't check again

A lot of people reporting bugs make the mistake of report-and-run. You should always be prepared to follow up. Also, if you're not already subscribed to an issue on here, there is a "Subscribe" button in the sidebar. Check the "Email" section of your account settings, etc.

> I don't have an example right now, but when one doesn't have javascript enabled for YT [so actually one cannot view the video embedded, one only see the link] in a page, one only can copy one version of the link sort of https...embeded....yt.com , no mouse over title option.

Not immediately clear why this doesn't work. Guess I'll look into it.

> I mean sort of convert https://web.archive.org/web/20190222375734/https://example.com/article/ into https://web.archive.org/web/https://example.com/article/ , which eg if bookmarked [or opened from copied link] would produce the last version of the page, maybe more recent than the date taken out.

Didn't know about that. Seems useful.

**Comment by ddawson** [comment]
Looks like the problem with embedded videos was just not having the content scripts running in them (technical: in this add-on, they're used both to display thumbnails and to handle copying to the clipboard). That was easy to fix. Still doesn't work if the video is added through a script, but that's a limitation of context menu handling in WebExtensions, as far as I can tell.

BTW, just for future reference, it wasn't really necessary to open a new issue. You can still comment on closed issues, and they can be reopened at any time.

**Comment by gitut2007** [comment]
> doesn't work if the video is added through a script

Ok, this is an example, [javascript disabled for the page, 'frame' for YT NOT allowed using umatrix: https://www.motorsportmagazine.com/news/sports-cars/video-highlights-may-7 
the eg 1st video I cannot view the video, but can see the frame with a little square and the link at the top-left, can copy this link using regular FF right-click option: 
https://www.youtube.com/embed/c4-AQF9KC88

With Link Tools addon, I would like to have option to copy a https://youtu.be/... or a regular YT link, but even tough I see many options, none copy anything.

 _Just in case it's something different_ to 'if the video is added through a script'.

Thx.

**Comment by ddawson** [comment]
> using umatrix

This is a pretty important detail. I tried it out, and it's blocking the contents of the iframe that holds the video, replacing it with its own inline content as a `data:` URI. WebExtensions don't seem to like to run content scripts in such pages, so I can't copy from those. Or at least, I usually can't. Maybe I should find a way around all of that. But in the meantime, I fixed a related problem of "`https://www.youtube.com/embed/`..." not being recognized. I overlooked that one.

Anyway, if you don't mind whitelisting YouTube, you can try that out. Maybe click on the jigsaw icon, and next to "YouTube as 3rd-party" (on the right), click the "import" icon; if you do it under the * scope, you can whitelist it on all sites. You might need to use the reload button in the uMatrix popup to properly see the changes for a currently loaded site. After that, you should see YouTube's normal embed interface.

**Comment by gitut2007** [comment]
I prefer the pages lighter without loading YT frames, and then if I choose open a new tab and watch the video or just copy the link for later.
Anyway, thx.

**Comment by ddawson** [comment]
Turns out, it wasn't about scripts, AFAICT. The way I was trying to select text in order to copy it doesn't work if the action is initiated from an iframe, for some bizarre reason. However, I can now just use the new Clipboard API to directly copy to clipboard (one line of code instead of about 15, and not involving a content script), and not have to deal with that stuff. This really should work now no matter what frame the link might be in. It certainly works in the example you gave, as well as some others I found. It does, however, require at least Fx 63, so it wasn't possible last year.
